### PR TITLE
Fix NPE in PojoBuilderHelper where static methods follow bean convention

### PR DESCRIPTION
**Pull request by rozza**
JAVA-4119

### CODE REVIEW

**Review by jyemin — APPROVED** [review]
LGTM

**Comment by stIncMale** [comment]
The reason for this [Evergreen failure](https://spruce.mongodb.com/version/609a78eb32f4173cd1678d27/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&variant=tests-jdk8-unsecure__version~latest_os~linux_topology~sharded-cluster_auth~noauth_ssl~nossl_jdk~jdk8) is that [`com.mongodb.reactivestreams.client.ChangeStreamsTest` hanged](https://evergreen.mongodb.com/lobster/evergreen/task/mongo_java_driver_tests_jdk8_unsecure__version~latest_os~linux_topology~sharded_cluster_auth~noauth_ssl~nossl_jdk~jdk8_test_patch_ddefdb0cc29541f82d507318ef7f94ec03efcd39_609a78eb32f4173cd1678d27_21_05_11_12_30_36/0/task#bookmarks=0%2C92632%2C92767&l=1&shareLine=92632).

It's [not the first time](https://spruce.mongodb.com/version/6099561157e85a3ca91872b0/tasks?sorts=STATUS%3AASC%3BBASE_STATUS%3ADESC&variant=tests-jdk8-unsecure__version~latest_os~linux_topology~replicaset_auth~noauth_ssl~nossl_jdk~jdk8) ([logs](https://evergreen.mongodb.com/lobster/evergreen/task/mongo_java_driver_tests_jdk8_unsecure__version~latest_os~linux_topology~replicaset_auth~noauth_ssl~nossl_jdk~jdk8_test_patch_ddefdb0cc29541f82d507318ef7f94ec03efcd39_6099561157e85a3ca91872b0_21_05_10_15_49_38/0/task#bookmarks=0%2C81188&l=1&shareLine=81040)), something is not right, but it is not related to the changes in this PR.

File: bson/src/test/unit/org/bson/codecs/pojo/entities/SimpleWithStaticModel.java [hunk 1]
```diff
@@ -0,0 +1,90 @@
...
+            return false;
+        }
+
+        SimpleWithStaticModel that = (SimpleWithStaticModel) o;
+
+        if (getIntegerField() != null ? !getIntegerField().equals(that.getIntegerField()) : that.getIntegerField() != null) {
+            return false;
+        }
+        if (getStringField() != null ? !getStringField().equals(that.getStringField()) : that.getStringField() != null) {
+            return false;
```
[the remark this answers is not in the document]
Comment (rozza) [reply bson/src/test/unit/org/bson/codecs/pojo/entities/SimpleWithStaticModel.java:70]: Will update and check my intellij auto gen settings.

File: bson/src/test/unit/org/bson/codecs/pojo/entities/SimpleWithStaticModel.java [hunk 2]
```diff
@@ -0,0 +1,90 @@
...
+        }
+
+        return true;
+    }
+
+    @Override
+    public int hashCode() {
+        int result = getIntegerField() != null ? getIntegerField().hashCode() : 0;
+        result = 31 * result + (getStringField() != null ? getStringField().hashCode() : 0);
+        return result;
```
[the remark this answers is not in the document]
Comment (rozza) [reply bson/src/test/unit/org/bson/codecs/pojo/entities/SimpleWithStaticModel.java:80]: Will update and check my intellij auto gen settings.

**Pull request closed without merging**